### ISSUE TITLE
Table traits don't apply when the initial condition is a vector

### ISSUE THREAD

**Issue by ChrisRackauckas** [SciML/DifferentialEquations.jl#712]
This is an odd one.

```julia
using DifferentialEquations
using DataFrames
test = (du,u,p,t) -> du .= 1.01u

# Fails
prob = ODEProblem(test, rand(2), (-6.,100.))
sol = solve(prob)
df = DataFrame(sol)

#works
prob = ODEProblem(test, rand(2,2), (-6.,100.))
sol = solve(prob)
df = DataFrame(sol)
```

Seems like an upstream Tables.jl issue?

**Comment by ChrisRackauckas** [comment]
From @bkamins on the Slack:

```
I am assuming you are on DataFrames 0.22 and Tables 1.2.2. Start Julia with depwarn=yes. The reason is that temporarily - in DataFrames.jl 0.22 - we keep old code for AbstractMatrix handling, and sol is an AbstractMatrix (in 1.0 which, will be released soon, this deprecation will be removed - if something is a Tables.jl table it will be treated as such even if it is an AbstractMatrix).
Additionally - your code has a problem - not related with DataFrames.jl, but Tables.jl integration. In the case that sol has three dimensions then e.g. Tables.columns(sol) works correctly. Wihle when sol is two dimensional you get an error than sol is not a table.
In conclusion:
till DataFrames.jl is released the AbstractMatrix case is special cased, and you have to write DataFrames(Tables.columns(sol)) to have things working under DataFrames.jl 0.22 (under DataFrames.jl 1.0 just what you have will work);
In order for things to work you should make your sol compliant with Tables.jl API. The problem is that Tables.columns does not allow AbstractMatrix as an argument by default (see line 5 in matrix.jl in Tables.jl implementation). This needs to be overriden for your type
```

**Comment by bkamins** [comment]
My earlier comment was an issue with the package dependencies and I did not have latest DiffEqBase.jl installed. Now I have a clean environment with the release from today. The status is (for the not-working version):
* `DataFrames(sol)` would not work under any version of DataFrames.jl in the past; it will work correctly under DataFrames.jl 1.0 when the deprecation introduced in 0.22 will take an effect (this change was made exactly to allow the kind of use case you want to work)
* For the time being use `Tables.columns(sol) |> DataFrame` and all will work correctly

---

By the way. I have noticed that your code does not handle 0-dimensional containers:
```
prob = ODEProblem(test, fill(rand()), (-6.,100.))
```
is this intended? (if it worked this could be a corner case for Tabes.jl integration, but currently it does not work so I cannot test it)

**Comment by ChrisRackauckas** [comment]
0-dimensional containers don't make too much sense so that's okay. I'll mark this as upstream then and it should fix itself with the DataFrames.jl 1.0 release. Thanks!

**Comment by bkamins** [comment]
@ChrisRackauckas - one more thing. On DataFrames.jl `main` things should work as expected already. Could you please double check? Thank you!

**Comment by ChrisRackauckas** [comment]
Main works. Awesome, can't wait! Thanks!
